src/org/etools/j1939_84/controllers/part01/Part01Step26Controller.java

```diff
@@ -300,47 +310,588 @@ protected void run() throws Throwable {
+                }
+                // b. For MY2022-23 HEV and BEV drives, Warn each PG query, where no
+                // response was
+                if (getEngineModelYear() >= 2022 && getEngineModelYear() <= 2023) {
+                    addWarning("6.1.26.22.b - No response was received from "
+                            + module.getModuleName() + "for PG "
+                            + pg);
+                }
+            } else {
+                hybridPacketForPg.getSpns()
+                                 .forEach(spn -> {
+                                     // 6.1.26.22.c. Fail each PG query where any accumulator
+                                     // value received is greater than FAFFh.
+                                     if (spn.getRawValue() >= 0xFAFFL) {
+                                         addFailure("6.1.26.22.c - Bin value received is greater than 0xFAFFFFFF(h) from "
+                                                 + module.getModuleName() + " for " + spn);
+                                     }
+                                 });
+            }
+        }
+    }
+
+    private void testSp12730(OBDModuleInformation module) {
+        // 6.2.17.11 Actions6 for all MY2022+ Engines
+        // 6.2.17.11.a - DS request messages to ECU that indicated support in DM24 for upon request SP 12730 (GHG
+        // Tracking Lifetime Engine Run Time) for PG 64252 GHG Tracking Lifetime Array Data.
+        var ghgTrackingLifetimePackets = requestPackets(module.getSourceAddress(),
+                                                        GHG_TRACKING_LIFETIME_PG)
+                                                                                  .stream()
+                                                                                  // 6.2.17.11.b - Record each value
+                                                                                  // for use in Part 12.
+                                                                                  .peek(this::save)
+                                                                                  .collect(Collectors.toList());
+
+        GenericPacket packetForPg = haveResponseWithPg(ghgTrackingLifetimePackets, GHG_TRACKING_LIFETIME_PG);
+        if (packetForPg == null) {
+            // 6.2.17.12.a. Fail PG query where no response was received
+            addFailure("6.1.26.12.a - No response was received from "
+                    + module.getModuleName() + "for PG "
+                    + GHG_TRACKING_LIFETIME_PG);
+        } else {
+            packetForPg.getSpns()
+                       .stream()
+                       .filter(spn -> {
+                           // FIXME: requirements need updated
+                           // @Joe this is implemented correctly. We just need to remove comment when new document
+                           // with correction is released.
+                           return spn.getRawValue() > 0xFAFFFFFFL;
```

Comment:
The error message indicates we should be comparing with 0xFAFF, but that's not what's used here.

Reply:
However, we have been asked to update that value to the correct one as noted in the comment.  I believe the requirement was corrected in the new document we received in May.  I updated the comments with the requirements.